include/mbedtls/ssl.h

```diff
@@ -1162,6 +1164,23 @@ struct mbedtls_ssl_session
     uint32_t MBEDTLS_PRIVATE(ticket_lifetime);   /*!< ticket lifetime hint    */
 #endif /* MBEDTLS_SSL_SESSION_TICKETS && MBEDTLS_SSL_CLI_C */
 
+#if defined(MBEDTLS_SSL_PROTO_TLS1_3) && defined(MBEDTLS_SSL_SESSION_TICKETS)
+    uint8_t MBEDTLS_PRIVATE(ticket_flags);                  /*!< Ticket flags */
+    uint32_t MBEDTLS_PRIVATE(ticket_age_add);               /*!< Randomly generated value used to obscure the age of the ticket */
+    uint8_t MBEDTLS_PRIVATE(key_len);                       /*!< PSK key length */
+
+#if defined(MBEDTLS_SHA384_C)
+    unsigned char MBEDTLS_PRIVATE(key)[48];                 /*!< key (48 byte) */
+#elif defined(MBEDTLS_SHA256_C)
+    unsigned char MBEDTLS_PRIVATE(key)[32];                 /*!< key (32 byte) */
+#endif /* MBEDTLS_SHA256_C  */
+
+#if defined(MBEDTLS_HAVE_TIME) && defined(MBEDTLS_SSL_CLI_C)
```

Comment:
If `MBEDTLS_HAVE_TIME` isn't defined, aren't there going to be other problems (e.g. what would you add `ticket_age_add` to?) beyond just `ticket_received`?

Reply:
I prefer keep it at this moment. 

I put this comment in #6087. I'd like think about it when whole new session ticket finished.